source/applications/dependency-injection.md

```diff
@@ -0,0 +1,183 @@
+Ember applications utilize the [dependency
+injection](https://en.wikipedia.org/wiki/Dependency_injection) ("DI") design
+pattern to declare and instantiate classes of objects and dependencies between
+them. Applications and application instances each serve a role in Ember's DI
+implementation.
+
+An `Ember.Application` serves as a "registry" for dependency declarations.
+Factories (i.e. classes) are registered with an application, as well as rules
+about "injecting" dependencies that are applied when objects are instantiated.
+
+An `Ember.ApplicationInstance` serves as a "container" for objects that are
+instantiated from registered factories. Application instances provide a means to
+"look up" (i.e. instantiate and / or retrieve) objects.
+
+> _Note: Although an `Application` serves as the primary registry for an app,
+each `ApplicationInstance` can also serve as a registry. Instance-level
+registrations are useful for providing instance-level customizations, such as
+A/B testing of a feature._
+
+## Factory Registrations
+
+A factory can represent any part of your application, like a _route_,
+_template_, or custom class. Every factory is registered with a particular key.
+For example, the index template is registered with the key `template:index`, and
+the application route is registered with the key `route:application`.
+
+Registration keys have two segments split by a colon (`:`). The first segment is
+the framework factory type, and the second is the name of the particular
+factory. Hence, the `index` template has the key `template:index`. Ember has
+several built-in factory types, such as `service`, `route`, `template`, and
+`component`.
+
+You can create your own factory type by simply registering a factory with the
+new type. For example, to create a `user` type, you'd simply register your
+factory with `application.register('user:user-to-register')`.
+
+Factory registrations must be performed either in application or application
+instance initializers (with the former being much more common).
+
+For example, an application initializer could register a `Logger` factory with
+the key `logger:main`:
+
+```app/initializers/logger.js
+export function initialize(application) {
```

Comment:
Exporting a bare function like this is OK? Doesn't this need the `Ember.Application.initializer(` call?